src/core/algorithms/md/hymd/preprocessing/similarity_measure/distance_similarity_measure.cpp

```diff
@@ -0,0 +1,108 @@
+#include <numeric>
+
+#include "algorithms/md/hymd/preprocessing/similarity_measure/distance_similarity_measure.h"
+#include "config/exceptions.h"
+#include "model/types/date_type.h"
+
+double NumberDifference(const std::byte* left, const std::byte* right) {
+    auto left_val = model::Type::GetValue<model::Double>(left);  
+    auto right_val = model::Type::GetValue<model::Double>(right);
+
+    return std::abs(static_cast<double>(left_val - right_val));
+}
+
+double DateDifference(const std::byte* left, const std::byte* right) {
+    auto left_date = model::Type::GetValue<model::Date>(left);
+    auto right_date = model::Type::GetValue<model::Date>(right);
+
+    model::DateType date_type;
+    return static_cast<double>(abs(date_type.SubDate(left, right).days()));
+}
+
+
+namespace algos::hymd::preprocessing::similarity_measure {
+
+using SimInfo = std::map<Similarity, indexes::RecSet>;
+
+indexes::ColumnMatchSimilarityInfo DistanceSimilarityMeasure::MakeIndexes(
+        std::shared_ptr<DataInfo const> data_info_left,
+        std::shared_ptr<DataInfo const> data_info_right,
+        std::vector<indexes::PliCluster> const* clusters_right, model::md::DecisionBoundary min_sim, bool is_null_equal_null) const {
+    std::vector<model::md::DecisionBoundary> decision_bounds;
+    indexes::SimilarityMatrix similarity_matrix;
+    indexes::SimilarityIndex similarity_index;
+    auto const& data_left_size = data_info_left->GetElementNumber();
+    auto const& data_right_size = data_info_right->GetElementNumber();
+    Similarity lowest = 1.0;
+    for (ValueIdentifier value_id_left = 0; value_id_left < data_left_size; ++value_id_left) {
+        std::vector<std::pair<Similarity, RecordIdentifier>> sim_rec_id_vec;
+        std::byte const* left_value = data_info_left->GetAt(value_id_left);
+        Similarity max_distance = 0;
+        std::vector<double> distances(data_info_right->GetElementNumber());
+
+        for (ValueIdentifier right_index = 0; right_index < data_right_size; ++right_index) {
+                std::byte const* right_value = data_info_right->GetAt(right_index);
+                Similarity distance = compute_similarity_(left_value, right_value);
+                distances[right_index] = distance;
+                max_distance = std::max(max_distance, distance);
+        }
+        auto get_similarity = [max_distance, &distances](ValueIdentifier value_id_right) {
+            if (max_distance == 0) return 1.0;
+            Similarity distance = distances[value_id_right];
+            return static_cast<Similarity>(max_distance - distance) / static_cast<Similarity>(max_distance);
+            };
+        for (ValueIdentifier value_id_right = 0; value_id_right < data_right_size;
+             ++value_id_right) {
+            Similarity similarity = get_similarity(value_id_right);
+            if (similarity < min_sim) {
+                // Metanome keeps the actual value for some reason.
+                lowest = 0.0 /*similarity???*/;
```

Comment:
I've already decided what to do with these, don't keep the comments.
```suggestion
                lowest = 0.0;
```